lib/src/tag/license_tags.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../model.dart';
+import '../third_party/spdx/licenses.dart';
+
+const _licenseFsfLibreTag = 'license:fsf-libre';
+const _licenseOsiApprovedTag = 'license:osi-approved';
+const _licenseUnknownTag = 'license:unknown';
+
+class LicenseTags {
+  final List<License> licenses;
+  final List<String> tags;
+  LicenseTags(this.licenses, this.tags);
+
+  late final isOnlyOsiApproved = tags.contains(_licenseOsiApprovedTag);
+  late final wasDetected = !tags.contains(_licenseUnknownTag);
+
+  late final osiApprovedLicenses =
+      licenses.where((l) => l.isOsiApproved).toList();
+  late final nonOsiApprovedLicenses =
+      licenses.where((l) => !l.isOsiApproved).toList();
+}
+
+LicenseTags createLicenseTags(List<License> licenses) {
```

Comment:
Can this be a constructor or static method on `LicenseTags`, like `LicenseTags.fromLicenses`